libs/geometry/src/TangentSpaceMesh.cpp

```diff
@@ -138,41 +132,35 @@ Algorithm selectAlgorithm(TangentSpaceMeshInput *input) {
             foundAlgo = true;
             break;
         case Algorithm::MIKKTSPACE:
-            if (IS_INPUT_TYPE(inputType, NORMALS_UVS_POSITIONS_INDICES)) {
+            if (isInputType(inputType, NORMALS_UVS_POSITIONS_INDICES)) {
                 outAlgo = Algorithm::MIKKTSPACE;
                 foundAlgo = true;
             }
             break;
         case Algorithm::LENGYEL:
-            if (IS_INPUT_TYPE(inputType, NORMALS_UVS_POSITIONS_INDICES)) {
+            if (isInputType(inputType, NORMALS_UVS_POSITIONS_INDICES)) {
                 outAlgo = Algorithm::LENGYEL;
                 foundAlgo = true;
             }
             break;
         case Algorithm::HUGHES_MOLLER:
-            if (IS_INPUT_TYPE(inputType, NORMALS)) {
+            if (isInputType(inputType, NORMALS)) {
                 outAlgo = Algorithm::HUGHES_MOLLER;
                 foundAlgo = true;
             }
             break;
         case Algorithm::FRISVAD:
-            if (IS_INPUT_TYPE(inputType, NORMALS)) {
+            if (isInputType(inputType, NORMALS)) {
                 outAlgo = Algorithm::FRISVAD;
                 foundAlgo = true;
             }
             break;
         case Algorithm::FLAT_SHADING:
-            if (IS_INPUT_TYPE(inputType, POSITIONS_INDICES)) {
+            if (isInputType(inputType, POSITIONS_INDICES)) {
                 outAlgo = Algorithm::FLAT_SHADING;
                 foundAlgo = true;
             }
             break;
-        case Algorithm::SIGN_OF_W:
-            if (IS_INPUT_TYPE(inputType, NORMALS_TANGENTS)) {
-                outAlgo = Algorithm::SIGN_OF_W;
-                foundAlgo = true;
-            }
-            break;
         default:
             PANIC_POSTCONDITION("Unknown algo %u", static_cast<uint8_t>(input->algorithm));
```

Comment:
That's actually not a post condition failure.  a post condition failure would be something like running out of memory, like... there was no mistake when the method was called, but the method is not able to do what it is supposed to do.
But like before I would argue that the `default` case cannot happen because we're switching on an enum.

Reply:
done